Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift

```diff
@@ -509,6 +509,30 @@ public extension TypeEffectSpecifiersSyntax {
   }
 }
 
+public extension WildcardPatternSyntax {
+  @available(*, deprecated, message: "remove 'typeAnnotation'")
+  init(
+    leadingTrivia: Trivia? = nil,
+    wildcard: TokenSyntax = .wildcardToken(),
+    typeAnnotation: TypeAnnotationSyntax? = nil,
+    trailingTrivia: Trivia? = nil
+  ) {
+    self.init(
+      leadingTrivia: leadingTrivia,
+      nil,
+      wildcard: wildcard,
+      nil,
+      trailingTrivia: trailingTrivia
+    );
+  }
+
+  @available(*, deprecated, message: "'typeAnnotation' was removed")
+  var typeAnnotation: TypeAnnotationSyntax? {
+    get { nil }
+    set {}
+  }
```

Comment:
I think we should also cover the unexpected children in the compatibility layer. See https://github.com/apple/swift-syntax/pull/2379#discussion_r1421197690